### ISSUE TITLE
ErrorBars for X and Y axis

### ISSUE THREAD

**Issue by KrisWragg** [EPPlusSoftware/EPPlus#394]
Hi

We have just recently licenced version 5 and I have been going through updating various chunks of our code after changing versions.

Previously we had several hacks to support error bars, but it seems there is now support for this in the library. I have the function below for ScatterChart to add error bars for both X and Y axis, but I cannot see how to implement this currently as you can only choose one or the other?

```
/// <summary>
/// This function will add both plus/minus error bars to a scatter chart, with different ranges for both X and Y directions
/// LIMITATIONS: Only adds Both error bars to both axis
/// </summary>
/// <param name="seriesIndex">The index of the series, zero based</param>
/// <param name="chart">The chart to add error bars to</param>
/// <param name="errorRange_X">Custom data to use for X axis</param>
/// <param name="errorRange_Y">Custom data to use for Y axis</param>
public static void AddErrorBars(ExcelScatterChart chart, int seriesIndex, string errorRange_X, string errorRange_Y)
{
	var nsmgr = new XmlNamespaceManager(chart.ChartXml.NameTable);
	nsmgr.AddNamespace("c", "http://schemas.openxmlformats.org/drawingml/2006/chart");
	nsmgr.AddNamespace("a", "http://schemas.openxmlformats.org/drawingml/2006/main");
	nsmgr.AddNamespace("r", "http://schemas.openxmlformats.org/officeDocument/2006/relationships");
	nsmgr.AddNamespace("c16r2", "http://schemas.microsoft.com/office/drawing/2015/06/chart");

	XNamespace ns_c = "http://schemas.openxmlformats.org/drawingml/2006/chart";
	XNamespace ns_a = "http://schemas.openxmlformats.org/drawingml/2006/main";

	var xDoc = XDocument.Parse(chart.ChartXml.InnerXml);

	var cSers = xDoc.XPathSelectElements("/c:chartSpace/c:chart/c:plotArea/c:scatterChart/c:ser", nsmgr).ToList();

	if (seriesIndex >= cSers.Count)
		return;

	cSers[seriesIndex].Add(
		#region X axis error bar
		new XElement(ns_c + "errBars",
			new XElement(ns_c + "errDir",
				new XAttribute("val", "x")
			),

			new XElement(ns_c + "errBarType",
				new XAttribute("val", "both")
			),

			new XElement(ns_c + "errValType",
				new XAttribute("val", "cust")
			),

			new XElement(ns_c + "noEndCap",
				new XAttribute("val", "0")
			),

			new XElement(ns_c + "plus",
				new XElement(ns_c + "numRef",
					new XElement(ns_c + "f", errorRange_X)
				)
			),

			new XElement(ns_c + "minus",
				new XElement(ns_c + "numRef",
					new XElement(ns_c + "f", errorRange_X)
				)
			)
		),
	#endregion

		#region Y axis error bar
		new XElement(ns_c + "errBars",
			new XElement(ns_c + "errDir",
				new XAttribute("val", "y")
			),

			new XElement(ns_c + "errBarType",
				new XAttribute("val", "both")
			),

			new XElement(ns_c + "errValType",
				new XAttribute("val", "cust")
			),

			new XElement(ns_c + "noEndCap",
				new XAttribute("val", "0")
			),

			new XElement(ns_c + "plus",
				new XElement(ns_c + "numRef",
					new XElement(ns_c + "f", errorRange_Y)
				)
			),

			new XElement(ns_c + "minus",
				new XElement(ns_c + "numRef",
					new XElement(ns_c + "f", errorRange_Y)
				)
			)
		)
		#endregion
	);

	chart.ChartXml.LoadXml(xDoc.ToString());
}
```

**Comment by JanKallman** [comment]
Hello,
Thanks for pointing this out. EPPlus don't support adding multiple error bars, which is a incorrect behavior for area, scatter and bubble charts. 
I will provide a fix to support the second error bar for these chart types asap. 
Sorry for the inconvenience

**Comment by KrisWragg** [comment]
@JanKallman that is great, thanks for quick response :)

The error bar support on other chart types has already allowed us to rip out large chunks of similar code, V5 is a great improvement over V4.

**Comment by JanKallman** [comment]
Great to hear. I spent yesterday to add better support for this, so I will hopefully have something to commit later today or tomorrow.

**Comment by JanKallman** [comment]
I have added a fix for this issue. Area-, Bubble- and Scatter chart series now has a new property ErrorBarsX that will be set by default by the AddErrorBars method. Feel free to try it using our develop nuget feed as described here https://github.com/EPPlusSoftware/EPPlus/wiki/Using-our-Develop-Nuget-Feed

**Comment by JanKallman** [comment]
Added/fixed in version 5.7.0
